### PR TITLE
Bookmark

### PR DESCRIPTION
**Pull request by tanishk15sharma**
- [x] Bookmark slice
- [x] Bookmark page
- [x] Bookmark REST API calls
- [x] Get bookmarked posts
- [x] Add post to bookmarks
- [x] Delete bookmarked post

### CODE REVIEW

File: src/components/BookmarksFeed.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React, { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { getAllBookmarks } from "../features/bookmarkSlice";
+import { PostCard } from "./PostCard";
+const BookmarksFeed = () => {
+  const dispatch = useDispatch();
+  const { bookmarks, loading } = useSelector((state) => state.bookmarks);
+  console.log(bookmarks);
```
Comment (Pratham82) [inline src/components/BookmarksFeed.js:8]: remove logs

File: src/features/bookmarkSlice.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+);
+const bookmarksSlice = createSlice({
+  name: "bookmarks",
+  initialState: {
+    loading: false,
+    bookmarks: [],
+  },
+  reducers: {
+    removePostFromBookmark: (state, { payload }) => {
+      console.log(payload);
```
Comment (Pratham82) [inline src/features/bookmarkSlice.js:57]: remove log

**Review by Pratham82 — APPROVED** [review]
1. Code structure is good,
2. Null checking for the components done is also good.
3. Add error handling try catch blocks in postSlice
4. Just a suggestion if you create folder for each feature like, bookmark then add bookmarkSlice and constants for the book, that will add more scalability for futures
5. You can also abstract out the repeated promises from the slice and add it to services.

**Review by Prabhav-Sharma — APPROVED** [review]
Clean code, learned about image hosting on Cloudinary.
Liked the usage of asyncThunk.
You can merge it.

**Pull request merged**